pkg/streamingpromql/limiting/memory_consumption.go

```diff
@@ -3,13 +3,39 @@
 package limiting
 
 import (
+	"context"
 	"sync"
 
 	"github.com/prometheus/client_golang/prometheus"
 
 	"github.com/grafana/mimir/pkg/util/limiter"
 )
 
+type contextKey int
+
+const (
+	memoryConsumptionTracker contextKey = 0
+)
+
+// FromContextWithFallback returns a MemoryConsumptionTracker that has been added to this
+// context. If there is no MemoryConsumptionTracker in this context, a new no-op tracker that
+// does not enforce any limits is returned.
+func FromContextWithFallback(ctx context.Context) *MemoryConsumptionTracker {
+	tracker, ok := ctx.Value(memoryConsumptionTracker).(*MemoryConsumptionTracker)
+	if !ok {
+		return NewMemoryConsumptionTracker(0, nil)
+	}
```

Comment:
[Nit] Can't you just check whether the returned `tracker` is nil? That's also more robust than just checking for existence:
```suggestion
	tracker := ctx.Value(memoryConsumptionTracker).(*MemoryConsumptionTracker)
	if tracker == nil {
		return NewMemoryConsumptionTracker(0, nil)
	}
```